### PR TITLE
Bash test user secrets

### PR DESCRIPTION
**Pull request by ycliuhw**
Add bash tests for user secrets.

### CODE REVIEW

File: tests/suites/secrets_iaas/juju.sh [hunk 1]
```diff
@@ -66,12 +66,75 @@ check_secrets() {
...
+	secret_uri=$(juju --show-log add-secret owned-by="$model_name-1" --info "this is a user secret")
+	secret_short_uri=${secret_uri##*:}
+
+	check_contains "$(juju --show-log show-secret "$secret_uri" --revisions | yq ".${secret_short_uri}.description")" 'this is a user secret'
+
+	# create a new revision 2.
+	juju --show-log update-secret "$secret_uri" --info info owned-by="$model_name-2"
+	check_contains "$(juju --show-log show-secret "$secret_uri" --revisions | yq ".${secret_short_uri}.description")" 'info'
+
+	# grant secret to the app, and now the application can access the revision 2.
```
Comment (wallyworld) [inline tests/suites/secrets_iaas/juju.sh:88]: Can we add a check that app cannot access the secret before the grant is made.

**Review by wallyworld — APPROVED** [review]
Thank you. I think we just need to add a check that no access is allowed before granting access.
Will drain tests come next?

**Comment by ycliuhw** [comment]
> Thank you. I think we just need to add a check that no access is allowed before granting access. Will drain tests come next?

Yeah, we will have more  tests for drain.

**Pull request merged**